script/domVisitor.ts

```diff
@@ -73,7 +80,8 @@ async function visitPages(
       const appendQueryString =
         queryString ? (archiveUrl ? `?archive=${archiveUrl}&${queryString}` : `?${queryString}`)
                     : archiveUrl ? `?archive=${archiveUrl}` : '';
-      await page.goto(`${rootUrl}${pageUrl}${appendQueryString}`);
+      const pageRedirection = checkIfPageHasRedirect(redirects, pageUrl);
+      await page.goto(`${rootUrl}${pageRedirection ? pageRedirection : pageUrl}${appendQueryString}`);
```

Comment:
You could change the `checkIfPageHasRedirect` a little bit and do something like this:
```suggestion
      const validatedUrl = getPageUrlOrRedirection(redirects, pageUrl);
      await page.goto(`${rootUrl}${validatedUrl}${appendQueryString}`);
```